cmd/controller/controller.go

```diff
@@ -27,19 +32,29 @@ import (
 )
 
 const (
-	defaultRepoURL = "https://kubernetes-charts.storage.googleapis.com"
-	maxRetries     = 5
+	defaultNamespace = "kube-system"
+	defaultRepoURL   = "https://kubernetes-charts.storage.googleapis.com"
+	maxRetries       = 5
 )
 
 // Controller is a cache.Controller for acting on Helm CRD objects
 type Controller struct {
 	queue      workqueue.RateLimitingInterface
 	informer   cache.SharedIndexInformer
+	kubeClient kubernetes.Interface
 	helmClient *helm.Client
 }
 
+type httpClient interface {
+	Do(req *http.Request) (*http.Response, error)
+}
+
+var netClient httpClient = &http.Client{
+	Timeout: time.Second * 10,
```

Comment:
A 10 second timeout is very aggressive, particularly considering we have to download some content, probably with an HTTPS handshake, and we have no fallback afaics - when we hit this timeout, we're just going to abort.